### PR TITLE
RSDK-4679 - Add from_dm_from_extra util function

### PR DESCRIPTION
**Pull request by cheukt**
grabs the value of `fromDataManagement` from the extra dict

### CODE REVIEW

**Review by dgottlieb — APPROVED** [review]

File: src/viam/sdk/common/utils.cpp [hunk 1]
```diff
@@ -130,5 +130,18 @@ bool operator==(const response_metadata& lhs, const response_metadata& rhs) {
     return lhs.captured_at == rhs.captured_at;
 }
 
+bool from_dm_from_extra(const AttributeMap& extra) {
+    if (!extra) {
+        return false;
+    }
+    if (auto pos = extra->find("fromDataManagement"); pos != extra->end()) {
```
Comment (acmorrow) [inline src/viam/sdk/common/utils.cpp:137]: I think this syntax is C++17 specific but the driver targets C++14.

File: src/viam/sdk/common/utils.cpp [hunk 2]
```diff
@@ -130,5 +130,18 @@ bool operator==(const response_metadata& lhs, const response_metadata& rhs) {
...
+        ProtoType value = *(pos->second);
+        const bool* boolValue = value.get<bool>();
+        return boolValue ? *boolValue : false;
+    } else {
+        return false;
```
Comment (acmorrow) [inline src/viam/sdk/common/utils.cpp:142]: This function has three explicit `return false` statements, but I think it only needs one:

```
if (extra) {
    if (auto pos ...) {
        ...
        if (boolValue) { return *boolValue; }
    }
}
return false;
```

File: src/viam/sdk/common/utils.cpp (hunk 1 above)
Comment (cheukt) [reply src/viam/sdk/common/utils.cpp]: thanks for the catch! fixed

File: src/viam/sdk/common/utils.cpp (hunk 2 above)
Comment (cheukt) [reply src/viam/sdk/common/utils.cpp]: changed the structure a little, but I do prefer using early returns to reduce nesting and make it easier to read

**Review by acmorrow — APPROVED** [review]
LGTM

**Review by stuqdog — APPROVED** [review]

**Pull request merged**